python/pyarrow/src/arrow/python/platform.h

```diff
@@ -32,5 +32,10 @@
 #if _MSC_VER >= 1900
 #undef timezone
 #endif
+
+// https://bugs.python.org/issue36020
+#if _MSC_VER >= 1500
+#undef snprintf
+#endif
```

Comment:
It seems that the macro was removed by https://github.com/python/cpython/commit/e822e37946f27c09953bb5733acf3b07c2db690f .

How about the following?

```cpp
#ifdef snprintf
#undef snprintf
#endif
```